build.gradle

```diff
@@ -150,6 +151,10 @@ subprojects { subproject ->
 			append = false
 			destinationFile = file("$buildDir/jacoco.exec")
 		}
+		if (isTravisBuild) {
+			// restrict memory usage, so tests don't fail with exit code 137 on travis
+			maxHeapSize = '256m'
```

Comment:
There are 2 different things: the main gradle process which does the build, it's affected by GRADLE_OPTS and another forked process which performs the tests (currently it's forked once and performs all the tests, if specifying the "forkEvery" parameter then fork terminates after that number of tests and a new one is forked).

Currently only the forked "Gradle Test Executor" experience problems, so this change is enough.